src/components/Authentication/LoginRedirect.js

```diff
@@ -1,105 +1,111 @@
+                fname,
+                lname,
+                admin
+            }
+            props.setUser(userObject)
+            props.history.push(redirectRoute);
+        }
+
+        const pollForAuthenticatedUser = () => {
+            Auth.currentAuthenticatedUser({ bypassCache: true })
+                .then(async authUser => {
+                    const { email } = authUser.attributes
+    
+                    // If biztech email, assume admin (no need for 'sign in')
+                    if (email.substring(email.indexOf("@") + 1, email.length) === 'ubcbiztech.com') {
+
+                        populateUserAndRedirect(authUser, '/', true)
+    
+                    }
+                    // If not biztech username (normal member)
+                    else {
+    
+                        const student_id = authUser.attributes['custom:student_id'];
+                        
+                        // Detect if "first time sign up" by checking if custom:student_id is saved in the user pool
+                        // If the user's student_id exists in the user pool, check if the user is registered in the database
+                        // There is a possibility that a user exists in the user pool but not the database
+                        if(student_id)  {
+                            try {
+                                // check database
+                                const user = await fetchBackend(`/users/${student_id}`, 'GET');
+                                clearTimeout(timeoutRedirect)
+                                props.setUser({ ...user, admin: false }) // save to redux
+                                props.history.push('/'); // Redirect to the 'user home' page
+                            } catch(err) {
+                                // if the user exists in the user pool, but not the database, remove the user pool's student_id
+                                if(err.status === 404) {
+                                    clearTimeout(timeoutRedirect)
                                     await Auth.updateUserAttributes(authUser, { 'custom:student_id': '' });
                                     authUser.attributes['custom:student_id'] = null;
 
-                                    this.populateUserAndRedirect(authUser)
+                                    populateUserAndRedirect(authUser, '/signup')
                                 } else {
-                                    console.log(err.status)
+                                    console.log('Encountered an error querying database!', err.status);
                                 }
-                            })
-                    } else {
-                        // "First time sign up" aka no student_id in user pool
-                        clearTimeout(this.timeoutRedirect)
-                        this.populateUserAndRedirect(authUser)
+                            }
+                        } else {
```

Comment:
not sure if we want to cover the end case when the user "doesnt exist in the user pool, but exists in database"

Reply:
nah